diff --git a/node_modules/onnxruntime-react-native/android/CMakeLists.txt b/node_modules/onnxruntime-react-native/android/CMakeLists.txt
index 2f814e8..db18147 100644
--- a/node_modules/onnxruntime-react-native/android/CMakeLists.txt
+++ b/node_modules/onnxruntime-react-native/android/CMakeLists.txt
@@ -87,6 +87,9 @@ set_target_properties(
              CXX_EXTENSIONS OFF
              POSITION_INDEPENDENT_CODE ON)

+# 16 KB page alignment for Android 15+ devices
+target_link_options(onnxruntimejsi PRIVATE "-Wl,-z,max-page-size=16384")
+
 find_library(log-lib log)

 target_link_libraries(
diff --git a/node_modules/onnxruntime-react-native/android/build.gradle b/node_modules/onnxruntime-react-native/android/build.gradle
index 41b4359..b5c4a66 100644
--- a/node_modules/onnxruntime-react-native/android/build.gradle
+++ b/node_modules/onnxruntime-react-native/android/build.gradle
@@ -1,4 +1,5 @@
 import java.nio.file.Paths
+import org.gradle.util.internal.VersionNumber
 
 buildscript {
   repositories {
@@ -242,9 +243,9 @@ dependencies {
   implementation "junit:junit:4.12"
 
   if (useQnn) {
-    extractLibs "com.microsoft.onnxruntime:onnxruntime-android-qnn:latest.integration@aar"
+    extractLibs "com.microsoft.onnxruntime:onnxruntime-android-qnn:1.17.0@aar"
   } else {
-    extractLibs "com.microsoft.onnxruntime:onnxruntime-android:latest.integration@aar"
+    extractLibs "com.microsoft.onnxruntime:onnxruntime-android:1.17.0@aar"
   }
 
   if (VersionNumber.parse(REACT_NATIVE_VERSION) < VersionNumber.parse("0.71")) {
@@ -254,7 +255,7 @@ dependencies {
 
   // By default it will just include onnxruntime full aar package
   if (ortExtensionsEnabled) {
-    implementation "com.microsoft.onnxruntime:onnxruntime-extensions-android:latest.integration@aar"
+    implementation "com.microsoft.onnxruntime:onnxruntime-extensions-android:1.17.0@aar"
   }
 }
 
diff --git a/node_modules/onnxruntime-react-native/cpp/InferenceSessionHostObject.cpp b/node_modules/onnxruntime-react-native/cpp/InferenceSessionHostObject.cpp
index c8efd49..54fc484 100644
--- a/node_modules/onnxruntime-react-native/cpp/InferenceSessionHostObject.cpp
+++ b/node_modules/onnxruntime-react-native/cpp/InferenceSessionHostObject.cpp
@@ -219,7 +219,10 @@ DEFINE_GETTER(InferenceSessionHostObject::inputMetadata) {
         item.setProperty(runtime, "isTensor", Value(true));
 
         // symbolicDimensions
-        auto symbolicDimensions = tensorInfo.GetSymbolicDimensions();
+        size_t dim_count = tensorInfo.GetSymbolicDimensionsCount();
+        std::vector<const char*> symbolicDimensions(dim_count);
+        tensorInfo.GetSymbolicDimensions(symbolicDimensions.data(),
+                 symbolicDimensions.size());
         auto symbolicDimensionsArray =
             Array(runtime, symbolicDimensions.size());
         for (size_t j = 0; j < symbolicDimensions.size(); j++) {
@@ -282,7 +285,10 @@ DEFINE_GETTER(InferenceSessionHostObject::outputMetadata) {
         item.setProperty(runtime, "isTensor", Value(true));
 
         // symbolicDimensions
-        auto symbolicDimensions = tensorInfo.GetSymbolicDimensions();
+        size_t dim_count = tensorInfo.GetSymbolicDimensionsCount();
+        std::vector<const char*> symbolicDimensions(dim_count);
+        tensorInfo.GetSymbolicDimensions(symbolicDimensions.data(),
+                 symbolicDimensions.size());
         auto symbolicDimensionsArray =
             Array(runtime, symbolicDimensions.size());
         for (size_t j = 0; j < symbolicDimensions.size(); j++) {
